### PR TITLE
add sword charge labels

### PR DESCRIPTION
**Pull request by zelosos**
added sword charge max value as constant
label some charge related code

### CODE REVIEW

File: src/constants/gameplay.asm [hunk 1]
```diff
@@ -165,6 +165,9 @@ SWORD_ANIMATION_STATE_SWING_MIDDLE equ $03
 SWORD_ANIMATION_STATE_SWING_END    equ $04
 SWORD_ANIMATION_STATE_HOLDING      equ $05
 
+; Charge amount needed for spin attack
+MAX_SWORD_CHARGE equ $28
```
Comment (kemenaran) [inline src/constants/gameplay.asm:169]: This name could imply that this is the charge level for the sword throwing beams (when on full hearts). Maybe `SWORD_MAX_SPIN_CHARGE` would be a better name?

File: src/code/bank2.asm [hunk 2]
```diff
@@ -152,40 +152,40 @@ ENDC
...
     ld   a, [wIsShootingArrow]                    ; $4287: $FA $4C $C1
     and  a                                        ; $428A: $A7
-    jr   z, jr_002_4291                           ; $428B: $28 $04
+    jr   z, .jr_002_4291                           ; $428B: $28 $04
 
     dec  a                                        ; $428D: $3D
     ld   [wIsShootingArrow], a                    ; $428E: $EA $4C $C1
 
-jr_002_4291:
+.jr_002_4291:
```
Comment (kemenaran) [inline src/code/bank2.asm:160]: Thanks for turning all these labels into local labels 👍 

A small thing: technically putting a semicolon `:` after a label turn the label into an "exported" label. In this project this doesn't matter much: we export all labels anyway, and rgbds will give [higher priority](https://rednex.github.io/rgbds/rgbasm.5.html#SYMBOLS) to the dot, meaning the label will indeed be local.

But still, in this project we usually remove all semicolons after a local label.

If you feel like doing the relevant changes in a next PR (I'm fine with merging this one as-this), that'll be awesome :)

File: src/constants/gameplay.asm (hunk 1 above)
Comment (zelosos) [reply src/constants/gameplay.asm:169]: I do not understand. When is the sword throwing beams?
New name is god.

File: src/code/bank2.asm (hunk 2 above)
Comment (zelosos) [reply src/code/bank2.asm:160]: OK did not know that. Will make a new PR.

File: src/constants/gameplay.asm (hunk 1 above)
Comment (kemenaran) [reply src/constants/gameplay.asm:169]: The level 2 sword (the one found in the Search Seashells Mansion) throws beams when all hearts are full :)

Comment (zelosos) [reply src/constants/gameplay.asm:169]: I looked over all usages of wSwordCharge.
I think the max charge for this beams is the same value. $28

Comment (zelosos) [reply src/constants/gameplay.asm:169]: So rename to SWORD_MAX_SPIN_CHARGE?

**Pull request merged**